api/v1alpha4/common_types.go

```diff
@@ -101,11 +101,8 @@ var (
 
 const (
 	// MachineNodeNameIndex is used by the Machine Controller to index Machines by Node name, and add a watch on Nodes.
+	// Deprecated: Use api/v1alpha4/index.MachineNodeNameIndex instead.
 	MachineNodeNameIndex = "status.nodeRef.name"
-
-	// MachineProviderIDIndex is used to index Machines by ProviderID. It's useful to find Machines
-	// in a management cluster from Nodes in a workload cluster.
-	MachineProviderIDIndex = "spec.providerID"
```

Comment:
Should this be deprecated and removed in the next API version? Otherwise this is a breaking change...

Reply:
You are correct.  This should also be deprecated like `MachineNodeNameIndex`. Missed this change. Will add it back with a deprecation comment.